pkg/config/config.go

```diff
@@ -148,14 +188,23 @@ func (config *Config) GardenNames() []string {
 	return names
 }
 
-// Garden returns a Garden cluster from the list of configured Gardens
+// Garden returns the first matching Garden cluster by name (identity or alias) from the list
+// of configured Gardens
 func (config *Config) Garden(name string) (*Garden, error) {
-	i, ok := config.IndexOfGarden(name)
-	if !ok {
+	var gardenConfig *Garden
+
+	for idx := range config.Gardens {
+		cfg := &config.Gardens[idx]
+		if name == cfg.Name || name == cfg.Alias {
```

Comment:
not sure where to put this comment:

```
gardenctl target --garden dev --project myproject --shoot mycluster
Successfully targeted shoot "mycluster"
```

```bash
cat $KUBECONFIG

apiVersion: v1
clusters:
- cluster:
    extensions:
    - extension:
        gardenClusterIdentity: dev
        shootRef:
          name: mycluster
          namespace: garden-myproject
      name: client.authentication.k8s.io/exec
    server: https://api.example.com
# ...
```

the `gardenClusterIdentity` contains the alias, not the identity. Either we update `gardenlogin` to also support the alias, or we should only set the identity here

Reply:
I will have a look into this. Imho it would be better/cleaner to always use the identity here.